finagle-core/src/main/scala/com/twitter/finagle/loadbalancer/Sticky.scala

```diff
@@ -0,0 +1,95 @@
+package com.twitter.finagle.loadbalancer
+
+import com.twitter.finagle.{ClientConnection, NoBrokersAvailableException, Service,
+  ServiceFactory, ServiceFactoryProxy, Status}
+import com.twitter.finagle.service.FailingFactory
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.util.{Activity, Future, Promise, Time}
+
+/**
+ * A simple sticky load balancer which selects the first backend it
+ * can connect to and sticks to it unless that backend is not functioning
+ * properly, in which case it will change to another functioning backend.
```

Comment:
let's also add a note that if many nodes go down, and then later are revived, but the old nodes never go down, there's no mechanism to later move load back to the revived nodes.